File: server/src/main/java/de/litexo/model/internal/InternalOpenttdServerConfig.java
package de.litexo.model.internal;

import de.litexo.model.external.OpenttdServer;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class InternalOpenttdServerConfig {
    private String path;
    private int autoSaveMinutes = 5;

    // sha256 hash of the admin password

    private String passwordSha256Hash;

    List<OpenttdServer> servers = new ArrayList<>();
}

File: ui/src/app/api/models/internal-openttd-server-config.ts
/* tslint:disable */
/* eslint-disable */
import { OpenttdServer } from './openttd-server';
export interface InternalOpenttdServerConfig {
  autoSaveMinutes?: number;
  numberOfAutoSaveFilesToKeep?: number;
  numberOfManuallySaveFilesToKeep?: number;
  passwordSha256Hash?: string;
  path?: string;
  servers?: Array<OpenttdServer>;
}
